### PR TITLE
Remove references to CsvFileTests

### PR DESCRIPTION
**Pull request by luis-pabon-tf**
This PR handles some tech debt as a follow-up to PR #7384. We remove one of the enum values that is no longer needed, and references related to them.

The `CsvFileTests` topic usage was a bit of a "closed circle". The only non-test code touching it consisted of private vals, code only used by very specific tests, and an enum that has no text matches in the entire codebase for the removed value.

Upon making the changes, running various manual submissions into the pipeline and running all automated tets, I determined the removal was safe.

Test Steps:
1. Verify building and running the app works
2. Runs `test`, `testSmoke`, `testIntegration`

## Changes
- Deleted the `CsvFileTests` value from its enum
- Removed functions that exclusively tested `CsvFileTests`
- Remove schemas and test files only used with csv file tests

## Linked Issues
- #7415

### CODE REVIEW

**Review by jimmyfagan — COMMENTED** [review]
These changes look fine, but I think we're losing some valuable test coverage by deleting `CsvFileTests.kt`. Instead of deleting those files referencing the `CsvFileTests` topic, can we just change the topic used by those tests to be `test` or `covid-19`? Let me know if you want to talk about it.

**Comment by luis-pabon-tf** [comment]
> These changes look fine, but I think we're losing some valuable test coverage by deleting `CsvFileTests.kt`. Instead of deleting those files referencing the `CsvFileTests` topic, can we just change the topic used by those tests to be `test` or `covid-19`? Let me know if you want to talk about it.

I can do a checkup on coverage loss and move tests elsewhere while changing topics. Will do so as soon as I hit a good pausing point on my current task

**Comment by jimmyfagan** [comment]
Just to clarify, I don't necessarily mean code coverage, just that with these tests removed, we might not have any remaining tests using CSV input in our smoke tests. I did look at this a little bit on Friday, so when the time comes, let me know if you'd be interested in pairing!

File: prime-router/src/test/kotlin/CsvFileTests.kt [hunk 1]
```diff
@@ -108,7 +106,7 @@ class CsvFileTests {
         assertThat(settings.receivers).hasSize(2)
         assertThat(
             settings.findReceiver("federal-test.receiver")?.jurisdictionalFilter
-        ).isNotNull().hasSize(2)
+        ).isNotNull().hasSize(1)
```
Comment (jimmyfagan) [inline prime-router/src/test/kotlin/CsvFileTests.kt:109]: Why did this change?

Comment (luis-pabon-tf) [reply prime-router/src/test/kotlin/CsvFileTests.kt:109]: The tests on this file used the removed CSV defaults, so to keep it working I moved those defaults to be on federal test instead. The non-default juris filter for federal-test only has one criteria, which is why it's smaller.

You can see the relevant change on `test-organizations.yml`

**Review by arnejduranovic — APPROVED** [review]

**Pull request merged**